app/models/patient.rb

```diff
@@ -316,15 +317,20 @@ def latest_transfer
       ' && HOUR(CONVERT_TZ(?, "UTC", patients.time_zone)) <= ?) '\
       'OR (patients.preferred_contact_time IS NULL'\
       ' && HOUR(CONVERT_TZ(?, "UTC", patients.time_zone)) >= ?'\
-      ' && HOUR(CONVERT_TZ(?, "UTC", patients.time_zone)) <= ?)',
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::MORNING_CONTACT_WINDOW.first,
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::MORNING_CONTACT_WINDOW.last,
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::AFTERNOON_CONTACT_WINDOW.first,
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::AFTERNOON_CONTACT_WINDOW.last,
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::EVENING_CONTACT_WINDOW.first,
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::EVENING_CONTACT_WINDOW.last,
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::UNSPECIFIED_CONTACT_WINDOW.first,
-      Time.now.getlocal('-00:00'), PatientDetailsHelper::UNSPECIFIED_CONTACT_WINDOW.last
+      ' && HOUR(CONVERT_TZ(?, "UTC", patients.time_zone)) <= ?)'\
+      'OR (patients.preferred_contact_time IN (?)'\
+      ' && HOUR(CONVERT_TZ(?, "UTC", patients.time_zone)) >= CONVERT(patients.preferred_contact_time, UNSIGNED INTEGER)'\
+      ' && HOUR(CONVERT_TZ(?, "UTC", patients.time_zone)) <= CONVERT(patients.preferred_contact_time, UNSIGNED INTEGER))',
```

Comment:
Are these last two lines comparing `<=` and `>=` the equivalent of just `HOUR(CONVERT_TZ(?, "UTC", patients.time_zone)) = CONVERT(patients.preferred_contact_time, UNSIGNED INTEGER)`.

This may be a concern considering that there is a 3-5 hour window for the standard contact window times, but having a custom preferred contact time set only provides an hour-long window for the reminder to be sent. I remember having discussions about how the 3-5 hour window helps cover the case where if something goes wrong for one hour that the send assessments job runs, then there are 2-4 more chances/hours that the issue could possibly be corrected and patients would still receive reminders for that day. The way the custom preferred time is written there would not be this kind of redundancy. 

I'm not sure if this is still a concern because the conversations I am remembering happened several months ago - I will leave that up to @holmesie to determine.

Reply:
That is correct, thanks for suggesting a simplified version. Your concern also makes sense, @holmesie should I increase this window to a 3-5 hour window?

Reply:
extended the window here: https://github.com/SaraAlert/SaraAlert/pull/1019/commits/907413a38333431753e55e9ab3762856d0f0ffb1